src/Symfony/Component/HttpKernel/Attribute/MapQueryParameter.php

```diff
@@ -24,12 +24,27 @@ final class MapQueryParameter extends ValueResolver
     /**
      * @see https://php.net/filter.filters.validate for filter, flags and options
      *
-     * @param string|null $name The name of the query parameter. If null, the name of the argument in the controller will be used.
+     * @param class-string $resolver The class name of the resolver to use
      */
     public function __construct(
+        /**
+         * The name of the query parameter. If null, the name of the argument in the controller will be used.
+         */
         public ?string $name = null,
+
+        /**
+         * The filter to pass to "filter_var()".
```

Comment:
you could add `@var FILTER_VALIDATE_*|FILTER_SANITIZE_*|null` to help static analysis tools

Reply:
That's better! However, PHPStorm highlights the first part with the following message: `Undefined class 'FILTER_VALIDATE_' `. It feels a bit weird to have this highlight, and I wasn't able to find other examples of this other than in Parameter bag with the following snippet:

```php
    /**
     * Filter key.
     *
     * @param int                                     $filter  FILTER_* constant
     * @param int|array{flags?: int, options?: array} $options Flags from FILTER_* constants
     *
     * @see https://php.net/filter-var
     */
```

Maybe we should add the same snippet here

Reply:
Neither [PHPStan](https://phpstan.org/r/0f0e2aea-cf8b-4f88-b380-477e41f9b3c4) nor [Psalm](https://psalm.dev/r/33a8e3a779) recognizes the `@var FILTER_VALIDATE_*|FILTER_SANITIZE_*|null`. Go for your suggestion.

You can rise an issue on Psalm/PHPStan/PHPStorm to support global const pattern.

Reply:
it actually requires braces to work inside union types: https://phpstan.org/r/99fe52c1-ffca-4485-9386-a9a9a8f36211

Reply:
Works like a charm, thanks!